autoPyTorch/pipeline/components/training/trainer/__init__.py

```diff
@@ -249,15 +249,15 @@ def _fit(self, X: Dict[str, Any], y: Any = None, **kwargs: Any) -> 'TrainerChoic
         )
 
         # Support additional user metrics
-        additional_metrics = X['additional_metrics'] if 'additional_metrics' in X else None
-        if 'optimize_metric' in X:
-            additional_metrics = additional_metrics.append(X['optimize_metric']) if additional_metrics is not None \
-                else [X['optimize_metric']]
+        metrics = get_metrics(dataset_properties=X['dataset_properties'])
+        if 'additional_metrics' in X:
+            metrics.extend(get_metrics(dataset_properties=X['dataset_properties'], names=X['additional_metrics']))
+        if 'optimize_metric' in X and 'optimize_metric' not in [m.name for m in metrics]:
```

Comment:
Can you put a single-line comment for what `'optimize_metric' not in [m.name for m in metrics]` this means?